superchain/superchain.go

```diff
@@ -31,12 +31,23 @@ type BlockID struct {
 	Number uint64 `toml:"number"`
 }
 
+type OptimismConfig struct {
+	EIP1559Elasticity        uint64  `toml:"eip1559_elasticity" json:"eip1559Elasticity"`
+	EIP1559Denominator       uint64  `toml:"eip1559_denominator" json:"eip1559Denominator"`
+	EIP1559DenominatorCanyon *uint64 `toml:"eip1559_denominator_canyon,omitempty" json:"eip1559DenominatorCanyon,omitempty"`
+}
+type GenesisChainConfig struct {
+	// Optimism config, nil if not active
+	Optimism *OptimismConfig `toml:"optimism,omitempty" json:"optimism,omitempty"`
+}
+
 type ChainGenesis struct {
-	L1           BlockID      `toml:"l1"`
-	L2           BlockID      `toml:"l2"`
-	L2Time       uint64       `toml:"l2_time" json:"l2_time"`
-	ExtraData    *HexBytes    `toml:"extra_data,omitempty"`
-	SystemConfig SystemConfig `toml:"system_config" json:"system_config" `
+	L1           BlockID            `toml:"l1"`
+	L2           BlockID            `toml:"l2"`
+	L2Time       uint64             `toml:"l2_time" json:"l2_time"`
+	ExtraData    *HexBytes          `toml:"extra_data,omitempty"`
+	SystemConfig SystemConfig       `toml:"system_config" json:"system_config" `
+	Config       GenesisChainConfig `toml:"config" json:"config"`
```

Comment:
Why not put it as sub-attribute directly of `superchain.ChainConfig`? This config is the op-stack-specific subset of geth `ChainConfig` equivalent, so it would make sense to put it in the chain-config. It's going to change later, with additional attributes for new parameters, so I don't particularly like putting it into the `ChainGenesis` struct.

Reply:
Ok np I can un-nest it